feat: add endpoints for fetching and updating user data, add validati…

frooooooo7 opened the pull request:
…on rules

frooooooo7 committed:
feat: add endpoints for fetching and updating user data, add validation rules

marioooo0o committed:
Merge branch 'dev' into feat/profile_page

marioooo0o committed:
PHP Linting (Pint)

marioooo0o commented on app/Http/Requests/UserUpdateRequest.php:
```diff
@@ -0,0 +1,42 @@
+        return [
+            'name' => 'string|min:2|max:30|regex:/^[a-zA-ZÀ-ž\s\'-]+$/',
+            'last_name' => 'string|min:2|max:30|regex:/^[a-zA-ZÀ-ž\s\'-]+$/',
+            'email' => 'email|max:255',
```
dodaj tutaj również unique

marioooo0o commented on database/migrations/0001_01_01_000000_create_users_table.php:
```diff
@@ -14,9 +14,24 @@ public function up(): void
+            $table->string('house_number')->nullable();
             $table->timestamp('email_verified_at')->nullable();
             $table->string('password');
+
+            $table->string('card_first_name')->nullable();
+            $table->string('card_last_name')->nullable();
+            $table->string('card_number')->nullable();
+            $table->string('card_expiry_date')->nullable();
+            $table->string('card_cvv')->nullable();
```
Przenieść do oddzielenej tabeli z relacją 1:1

marioooo0o replied:
Zadbać o odpowiednie przechowywanie danych, ich zwracanie oraz szyfrowanie

marioooo0o commented on app/Http/Controllers/AuthController.php:
```diff
@@ -67,4 +69,24 @@ public function login(LoginRequest $request): JsonResponse
+        {
+            return response()->json(['message' => "Failed to update user's profile!"], Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+        return response()->json(['success' => true], Response::HTTP_OK);
```
Użycie struktury stworzonej w Course

frooooooo7 committed:
refactor:
- change structures in controllers
- card informations move to seperated table
- ensure that data is properly stored, returned and encrypted

frooooooo7 committed:
PHP Linting (Pint)

marioooo0o commented on app/Http/Requests/UpdateUserRequest.php:
```diff
@@ -0,0 +1,41 @@
+        return [
+            'name' => 'string|min:2|max:30|regex:/^[a-zA-ZÀ-ž\s\'-]+$/',
+            'last_name' => 'string|min:2|max:30|regex:/^[a-zA-ZÀ-ž\s\'-]+$/',
+            'email' => 'email|max:255|unique:users,email',
```
Sprawdz proszę jak zachowa się kontroler kiedy podasz ten sam email w update

marioooo0o commented on app/Http/Resources/CreditCardResource.php:
```diff
@@ -0,0 +1,36 @@
+        return str_repeat('#', strlen($cardNumber) - 4).substr($cardNumber, -4);
+    }
+
+    private function maskCvv(string $cvv): string
+    {
+        return str_repeat('*', strlen($cvv));
+    }
```
Czy cvv nie powinien zachowywać się podobnie jak hasło, tzn ze nie jest dostępne w resource, a przy potwierdzaniu transakcji podawany jest cvv i sprawdzany hash

frooooooo7 committed:
fix: in UpdateUserRequest unique email

frooooooo7 committed:
PHP Linting (Pint)

frooooooo7 committed:
fix:
